app/src/main/java/org/mozilla/focus/activity/ScreenNavigator.java

```diff
@@ -40,9 +70,25 @@ public static ScreenNavigator get(Context context) {
         }
     }
 
-    ScreenNavigator(MainActivity activity, MainMediator mediator) {
-        this.mainMediator = mediator;
-        this.browserMediator = new BrowserMediator(activity, mainMediator);
+    ScreenNavigator(@Nullable MainActivity activity) {
+        if (activity == null) {
```

Comment:
Is it valid to create an empty ScreenNavigator with no mediator..? Maybe throwing Exception here?

Reply:
On debug version, this will indeed lead to crash when the static factory of ScreenNavigator is called.
I allowed null here to make it easier to write a no-op version subclass.

Reply:
I'll see if i can improve this on another PR